src/validate.py

```diff
@@ -144,3 +126,53 @@ def parse_accept_language(header, supported_langs):
 
     # If all else fails, default the language.
     return DEFAULT_LANGUAGE.lang_code
+
+def get_json_files_from_dir(path):
+
+    files_found = []
+    for root, directories, files in os.walk(path):
+        for file in files:
+            if '.json' in file:
+                files_found.append(os.path.join(root, file))
+
+    return files_found
+
+def get_entries_from_json(path, p_key, s_key):
+
+    with open(path) as json_file:
+        data = json.load(json_file)
+
+    entries = []
+
+    if p_key in data:
+        for values in data.get(p_key):
+             entries.append(values.get(s_key))
+
+    return entries
+
+def get_chapters_for_year(file):
+
+    chapters = []
+
+    data = get_entries_from_json(file,'outline','chapters')
+    for list in data:
+        for entry in list:
+            chapters.append(entry.get('title').lower().replace(" ", "-").replace("/",""))
+
+    return chapters
+
+def get_parts_for_year(file):
+
+    parts = get_entries_from_json(file,'outline','part')
+
+    return parts
+
+json_files_present = get_json_files_from_dir(CONFIG_DIR)
+
+
+#TO DO - Lazy load previous year chapters
```

Comment:
What do you mean "lazy load"? Don't we need all valid chapters at each call to be able to validate any URL?

Reply:
I mean we can load all the chapters from the current year when the server starts up and for previous years we delay loading of chapters until we receive the first request. Or should we load everything right from the start ?

Reply:
OK. Not sure how gcloud works to be honest - is the python loaded once per request? Or initialised once and then can call extra things only when needed as you suggest? One for @rviscomi / @mikegeyser to answer I think.

Reply:
have refactored the code to load everything upfront.